### PR TITLE
487 adaptive tessellation component

### PR DESCRIPTION
**Pull request by mtao**
Resurrects the adaptive tessellation component that uses the image / displacement map functionality of the library

### CODE REVIEW

**Comment by mtao** [comment]
@daniel-zint i think you took over trying to get this to build or whatnot - are you working on another branch / what is the progress?

**Comment by daniel-zint** [comment]
> @daniel-zint i think you took over trying to get this to build or whatnot - are you working on another branch / what is the progress?

I still need to fix compilation issues. Now I am fighting with warnings on Mac but I hope that I will get it done on Monday so that it can be merged.

**Comment by daniel-zint** [comment]
Subsequent issue: #533

File: components/wmtk_components/adaptive_tessellation/src/wmtk/components/adaptive_tessellation/image/load_image_exr.cpp [hunk 1]
```diff
@@ -325,12 +324,13 @@ auto load_image_exr_split_3channels(const std::filesystem::path& path) -> std::
...
-            std::move(index_red),
-            std::move(index_green),
-            std::move(index_blue),
-            std::move(data_r),
-            std::move(data_g),
-            std::move(data_b)};
+    return {
+        static_cast<size_t>(exr_image.width),
+        static_cast<size_t>(exr_image.height),
+        std::move(index_red),
```
Comment (mtao) [inline components/wmtk_components/adaptive_tessellation/src/wmtk/components/adaptive_tessellation/image/load_image_exr.cpp:330]: lol i know you didn't write this but std::move on primitives doesn't do anything but potentially break optimizers

File: components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+#define STB_IMAGE_IMPLEMENTATION
+#include <stb_image.h>
+
+namespace wmtk::components::internal {
+
+Eigen::Matrix<long, -1, -1> tags_from_image(const std::filesystem::path& file)
+{
+    int width, height, channels;
+    unsigned char* img = stbi_load(file.string().c_str(), &width, &height, &channels, 0);
```
Comment (mtao) [inline components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp:13]: who takes responsibility for closing this? i suspect there's a free at the end of this file

File: components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    if (img == NULL) {
+        throw std::runtime_error("Error in loading the image");
+    }
+    const size_t img_size = width * height * channels;
+
+    Eigen::Matrix<long, -1, -1> pixel_matrix;
+    pixel_matrix.resize(height, width);
+
+    for (size_t i = 0; i < height; ++i) {
```
Comment (mtao) [inline components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp:22]:
```
pixel_matrix = Eigen::Map<wmtk::MatrixX<char>>(img,width,height);
```

should do this (perhaps with some extra stride info

File: components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp (hunk 2 above)
Comment (mtao) [inline components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp:13]: assert(channels == 1)

File: components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    const std::filesystem::path& file,
+    const std::string& tag_name)
+{
+    std::shared_ptr<wmtk::TriMesh> m = std::make_shared<wmtk::TriMesh>();
+
+    Eigen::Matrix<long, -1, -1> img = tags_from_image(file);
+
+    std::cout << "img:\n" << img << std::endl;
+
+    auto lex_index_v = [&img](size_t i, size_t j) { return i * (img.cols() + 1) + j; };
```
Comment (mtao) [inline components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp:42]: it's eigen :( so everything should be int (cuz img.cols() returns an int. surprised you didn't get narrowing warnings and whatnot everywhere

**Review by mtao — COMMENTED** [review]
i think this looks ok to me now @daniel-zint as this is technically my PR you gotta accept it xD

File: components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp (hunk 2 above)
Comment (daniel-zint) [reply components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp:13]: #535

**Pull request merged**